### PR TITLE
Add worker_init_fn in torch dataloader

### PR DESCRIPTION
**Pull request by sanskar107**
1.) Added `worker_init_fn` in torch dataloader to fix randomization issue.
2.) Skip batch(training) element with no bounding box.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d-ml/254)

### CODE REVIEW

**Review by ssheorey — CHANGES_REQUESTED** [review]
Using `int(time())` to set the random seed will make the results not reproducible. Also, the seed will be the same if 2 or more workers start up in the same second. Can we use 
[`torch.utils.data.get_worker_info().seed`](https://pytorch.org/docs/stable/data.html#torch.utils.data.get_worker_info) instead?

**Comment by sanskar107** [comment]
> Using `int(time())` to set the random seed will make the results not reproducible. Also, the seed will be the same if 2 or more workers start up in the same second. Can we use
> [`torch.utils.data.get_worker_info().seed`](https://pytorch.org/docs/stable/data.html#torch.utils.data.get_worker_info) instead?

In `worker_init_fn=lambda x: np.random.seed(x + int(time.time()))`, `x` is the worker id, and the seed will be different for all processes in the same epoch. I included time to take care of seeds across different epochs.
`torch.utils.data.get_worker_info().seed` also repeats across different epochs, should I just add epoch number in seed instead?

**Comment by ssheorey** [comment]
> In `worker_init_fn=lambda x: np.random.seed(x + int(time.time()))`, `x` is the worker id, and the seed will be different for all processes in the same epoch. 

Agreed.

> `torch.utils.data.get_worker_info().seed` also repeats across different epochs, should I just add epoch number in seed instead?

Yes, that should work. Though can you check that `torch.utils.data.get_worker_info().seed` is the same across epochs? The documentation says that this is obtained from the RNG of the main process and the `worker_id`. If it's created by just asking the main process RNG for a random number, it should vary across epochs. Tanel P.'s blog also seems to imply this.

**Comment by sanskar107** [comment]
> Yes, that should work. Though can you check that `torch.utils.data.get_worker_info().seed` is the same across epochs? The documentation says that this is obtained from the RNG of the main process and the `worker_id`. If it's created by just asking the main process RNG for a random number, it should vary across epochs. Tanel P.'s blog also seems to imply this.

I've tried running the minimal example given in Tanel P's blog. It seems like `torch.utils.data.get_worker_info().seed` is exactly same for the same worker_id across epochs. But output from that dataloader is different across epochs even though the seed is same. I believe this may be risky if in some other situation RNG may give same numbers.

**Review by ssheorey — APPROVED** [review]
